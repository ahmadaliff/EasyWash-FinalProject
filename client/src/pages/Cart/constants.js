export const GET_CARTS = 'GET_CARTS';
export const SET_CARTS = 'SET_CARTS';
export const RESET_CARTS = 'RESET_CARTS';
export const CHANGE_QUANTITY = 'cart/CHANGE_QUANTITY';
export const DELETE_FROM_CART = 'cart/DELETE_FROM_CART';
export const ADD_TO_CART = 'cart/ADD_TO_CART';
