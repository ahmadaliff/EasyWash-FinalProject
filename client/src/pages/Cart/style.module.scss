@import '@styles/base';

.mainWrap {
  padding: 1rem;
  @include apply-at-md-size {
    padding: 1rem 5rem;
  }

  .header {
    @include directionX;
    justify-content: space-between;
    width: 100%;
    align-items: center;
    .backButton {
      margin-left: auto;
      background-color: var(--color-primary);
      border-radius: 10rem;
      padding: 0.25rem 1rem 0.25rem 1rem;
      text-transform: capitalize;
      font-size: medium;
      &:hover {
        background-color: var(--color-primary);
      }
    }
  }
  .card {
    margin: 1rem 0 1rem 0;
    box-shadow: 0 0 0.25rem gray !important;
  }
  .createButton {
    margin-left: auto;
    background-color: var(--color-primary);
    border-radius: 10rem;
    padding: 0.25rem 1rem 0.25rem 1rem;
    text-transform: capitalize;
    font-size: small;
    &:hover {
      background-color: var(--color-primary);
    }
  }
}
