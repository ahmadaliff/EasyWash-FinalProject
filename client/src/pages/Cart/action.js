import { ADD_TO_CART, CHANGE_QUANTITY, DELETE_FROM_CART, GET_CARTS, RESET_CARTS, SET_CARTS } from './constants';

export const actionGetCarts = () => ({
  type: GET_CARTS,
});

export const actionSetCarts = (data) => ({
  type: SET_CARTS,
  data,
});

export const actionResetCarts = () => ({
  type: RESET_CARTS,
});

export const actionChangeQuantity = (id, quantity) => ({
  type: CHANGE_QUANTITY,
  id,
  quantity,
});

export const actionAddToCart = (id, quantity) => ({
  type: ADD_TO_CART,
  id,
  quantity,
});

export const actionDeleteFromCart = (id) => ({
  type: DELETE_FROM_CART,
  id,
});
