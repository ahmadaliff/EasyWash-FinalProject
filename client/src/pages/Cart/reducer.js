import { produce } from 'immer';
import { RESET_CARTS, SET_CARTS } from './constants';

export const initialState = {
  carts: null,
};

export const storedKey = ['carts'];

export const cartReducer = (state = initialState, action) =>
  produce(state, (draft) => {
    switch (action.type) {
      case SET_CARTS:
        draft.carts = action.data;
        break;
      case RESET_CARTS:
        return initialState;
    }
  });

export default cartReducer;
