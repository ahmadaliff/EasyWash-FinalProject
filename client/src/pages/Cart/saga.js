import toast from 'react-hot-toast';
import { call, put, takeLatest } from 'redux-saga/effects';

import intlHelper from '@utils/intlHelper';

import { apiAddToCart, apiChangeQuantityCart, apiDeleteFromCart, apiGetCarts } from '@domain/api';

import { ADD_TO_CART, CHANGE_QUANTITY, DELETE_FROM_CART, GET_CARTS } from '@pages/Cart/constants';
import { setLoading, showPopup } from '@containers/App/actions';
import { actionGetCarts, actionResetCarts, actionSetCarts } from './action';

function* sagaGetCarts() {
  yield put(setLoading(true));
  try {
    const response = yield call(apiGetCarts);
    yield put(actionSetCarts(response.data));
  } catch (error) {
    if (error?.response?.status === 400 || error?.response?.status === 404) {
      toast.error(intlHelper({ message: error.response.data.message }));
    } else {
      yield put(showPopup());
    }
  }
  yield put(setLoading(false));
}

function* sagaAddToCart({ id }) {
  yield put(setLoading(true));
  try {
    const response = yield call(apiAddToCart, id);
    yield put(actionResetCarts());
    yield put(actionGetCarts());
    toast.success(intlHelper({ message: response.message }));
  } catch (error) {
    if (error?.response?.status === 400 || error?.response?.status === 404) {
      toast.error(intlHelper({ message: error.response.data.message }));
    } else {
      yield put(showPopup());
    }
  }
  yield put(setLoading(false));
}

function* sagaDeleteFromCart({ id }) {
  yield put(setLoading(true));
  try {
    const response = yield call(apiDeleteFromCart, id);
    yield put(actionResetCarts());
    yield put(actionGetCarts());
    toast.success(intlHelper({ message: response.message }));
  } catch (error) {
    if (error?.response?.status === 400 || error?.response?.status === 404) {
      toast.error(intlHelper({ message: error.response.data.message }));
    } else {
      yield put(showPopup());
    }
  }
  yield put(setLoading(false));
}

function* sagaChangeQuantity({ id, quantity }) {
  yield put(setLoading(true));
  try {
    yield call(apiChangeQuantityCart, { serviceId: id, quantity });
    yield put(actionResetCarts());
    yield put(actionGetCarts());
  } catch (error) {
    if (error?.response?.status === 400 || error?.response?.status === 404) {
      toast.error(intlHelper({ message: error.response.data.message }));
    } else {
      yield put(showPopup());
    }
  }
  yield put(setLoading(false));
}

export default function* cartsSaga() {
  yield takeLatest(GET_CARTS, sagaGetCarts);
  yield takeLatest(ADD_TO_CART, sagaAddToCart);
  yield takeLatest(DELETE_FROM_CART, sagaDeleteFromCart);
  yield takeLatest(CHANGE_QUANTITY, sagaChangeQuantity);
}
