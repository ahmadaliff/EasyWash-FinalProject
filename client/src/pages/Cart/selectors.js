import { createSelector } from 'reselect';
import { initialState } from '@pages/Cart/reducer';

const selectCartsState = (state) => state.carts || initialState;

export const selectMerchants = createSelector(selectCartsState, (state) => state.carts);
