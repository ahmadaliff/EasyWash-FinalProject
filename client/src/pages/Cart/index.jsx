import PropTypes from 'prop-types';
import { useEffect } from 'react';
import { FormattedMessage } from 'react-intl';
import { useNavigate } from 'react-router-dom';
import { connect, useDispatch } from 'react-redux';
import { createStructuredSelector } from 'reselect';

import { Button, Card, CardContent } from '@mui/material';
import { ArrowBack } from '@mui/icons-material';

import MerchantInfo from '@components/MerchantInfo';
import ServiceCart from '@components/ServiceCart';
import NoData from '@components/NoData';

import { actionGetCarts, actionResetCarts } from '@pages/Cart/action';
import { selectMerchants } from '@pages/Cart/selectors';
import { actionSetOrder } from '@pages/Order/action';

import classes from '@pages/Cart/style.module.scss';

const Cart = ({ carts }) => {
  const navigate = useNavigate();
  const dispatch = useDispatch();

  useEffect(() => {
    dispatch(actionGetCarts());
    return () => {
      if (carts) dispatch(actionResetCarts());
    };
    // eslint-disable-next-line react-hooks/exhaustive-deps
  }, [dispatch]);

  const getCartOrder = (services) => {
    const cart = [];
    services.forEach((service) => {
      service.Carts.forEach((cartObj) => {
        const obj = { ...cartObj, serviceName: service.name, servicePrice: service.price };
        cart.push(obj);
      });
    });
    return cart;
  };

  return (
    <main className={classes.mainWrap} data-testid="cart-wrap">
      <div className={classes.header}>
        <h3>
          <FormattedMessage id="app_cart_header" />
        </h3>
        <Button
          variant="contained"
          className={classes.backButton}
          startIcon={<ArrowBack />}
          onClick={() => navigate(-1)}
          data-testid="button-back"
        >
          <FormattedMessage id="app_back" />
        </Button>
      </div>
      <Card className={classes.card}>
        <CardContent className={classes.cardContent}>
          {carts?.map((merchant, key) => (
            <div key={key}>
              <MerchantInfo merchant={merchant} />
              {merchant?.Services?.map((service, k) => (
                <div key={k}>
                  <ServiceCart service={service} />
                </div>
              ))}
              <Button
                variant="contained"
                className={classes.createButton}
                data-testid="button-create"
                onClick={() => {
                  dispatch(actionSetOrder(getCartOrder(merchant.Services)));
                  navigate('/order');
                }}
              >
                <FormattedMessage id="app_create_order" />
              </Button>
            </div>
          ))}
          {carts?.length === 0 && <NoData />}
        </CardContent>
      </Card>
    </main>
  );
};
Cart.propTypes = {
  carts: PropTypes.array,
};
const mapStateToProps = createStructuredSelector({
  carts: selectMerchants,
});
export default connect(mapStateToProps)(Cart);
