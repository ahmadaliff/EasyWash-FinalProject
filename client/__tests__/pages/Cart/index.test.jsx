import Cart from '@pages/Cart';
import { fireEvent, render } from '@utils/testHelper';

let wrapper;

const mockNavigate = jest.fn();
const mockDispatch = jest.fn();

jest.mock('@components/MapLeaflet', () => () => <div>mapleafletMock</div>);

jest.mock('react-redux', () => ({
  ...jest.requireActual('react-redux'),
  useDipatch: () => mockDispatch,
}));

jest.mock('@pages/Cart/selectors', () => ({
  selectMerchants: () => [
    {
      Services: [{ name: 'service', price: 2000, Carts: [{ id: 1, quantity: 1 }] }],
      name: 'merchant',
      location: '{"lat": -6.224934545453234, "lng": 106.86266439802276 }',
      description: 'description',
      imagePath: null,
      userId: 2,
    },
  ],
}));

jest.mock('react-router-dom', () => ({
  useNavigate: () => mockNavigate,
}));

beforeEach(() => {
  wrapper = render(<Cart carts={[{ name: 'test' }]} />);
});

describe('Cart Page', () => {
  test('Correct render', () => {
    const { getByTestId } = wrapper;
    expect(getByTestId('cart-wrap')).toBeInTheDocument();
    expect(getByTestId('button-back')).toBeInTheDocument();
  });
  test('Should navigate back', () => {
    const { getByTestId } = wrapper;
    const button = getByTestId('button-back');
    fireEvent.click(button);
    expect(mockNavigate).toHaveBeenCalledWith(-1);
  });
  test('Should create order and go to order page', () => {
    const { getByTestId } = wrapper;
    const button = getByTestId('button-create');
    fireEvent.click(button);
    expect(mockNavigate).toHaveBeenCalledWith('/order');
  });
  test('Should match with snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
