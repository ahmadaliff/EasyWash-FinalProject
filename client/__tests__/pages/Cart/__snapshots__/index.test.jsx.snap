// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Cart Page Should match with snapshot 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <main
        class="mainWrap"
        data-testid="cart-wrap"
      >
        <div
          class="header"
        >
          <h3>
            Cart
          </h3>
          <button
            class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium backButton css-sghohy-MuiButtonBase-root-MuiButton-root"
            data-testid="button-back"
            tabindex="0"
            type="button"
          >
            <span
              class="MuiButton-startIcon MuiButton-iconSizeMedium css-1d6wzja-MuiButton-startIcon"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                data-testid="ArrowBackIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M20 11H7.83l5.59-5.59L12 4l-8 8 8 8 1.41-1.41L7.83 13H20v-2z"
                />
              </svg>
            </span>
            Kembali
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </button>
        </div>
        <div
          class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root card css-bhp9pd-MuiPaper-root-MuiCard-root"
        >
          <div
            class="MuiCardContent-root cardContent css-46bh2p-MuiCardContent-root"
          >
            <div>
              <div
                class="merchant"
                data-testid="merchant-info"
              >
                <div
                  class="imgWrap"
                >
                  <div
                    class="MuiAvatar-root MuiAvatar-square MuiAvatar-colorDefault avatar css-18gxrea-MuiAvatar-root"
                  >
                    m
                  </div>
                </div>
                <div>
                  <h4
                    class="merchantInfoHeader"
                  >
                    merchant
                  </h4>
                  <span
                    class="profileInfo"
                  >
                    <p
                      class="profileText"
                    >
                      description
                    </p>
                  </span>
                </div>
                <button
                  aria-label="Pesan"
                  class="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorInherit MuiIconButton-sizeMedium chatButton css-zylse7-MuiButtonBase-root-MuiIconButton-root"
                  data-mui-internal-clone-element="true"
                  tabindex="0"
                  type="button"
                >
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                    data-testid="AddCommentOutlinedIcon"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M22 4c0-1.1-.9-2-2-2H4c-1.1 0-2 .9-2 2v12c0 1.1.9 2 2 2h14l4 4V4zm-2 13.17L18.83 16H4V4h16v13.17zM13 5h-2v4H7v2h4v4h2v-4h4V9h-4z"
                    />
                  </svg>
                  <span
                    class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                  />
                </button>
              </div>
              <div>
                <div
                  class="service"
                  data-testid="service-cart"
                >
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                    data-testid="DryCleaningIcon"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M19.56 11.36 13 8.44V7c0-.55-.45-1-1-1s-1-.45-1-1 .45-1 1-1 1 .45 1 1h2c0-1.84-1.66-3.3-3.56-2.95-1.18.22-2.15 1.17-2.38 2.35-.3 1.56.6 2.94 1.94 3.42v.63l-6.56 2.92c-.88.38-1.44 1.25-1.44 2.2v.01C3 14.92 4.08 16 5.42 16H7v6h10v-6h1.58c1.34 0 2.42-1.08 2.42-2.42v-.01c0-.95-.56-1.82-1.44-2.21zM18.58 14H17v-1H7v1H5.42c-.23 0-.42-.19-.42-.43 0-.17.1-.32.25-.38l6.75-3 6.75 3c.15.07.25.22.25.39 0 .23-.19.42-.42.42z"
                    />
                  </svg>
                  <div
                    class="serviceInfoWrap"
                  >
                    <div
                      class="serviceInfo"
                    >
                      <p>
                        service
                      </p>
                      <p>
                        RpÂ 2.000
                      </p>
                    </div>
                    <div
                      class="quantity"
                    >
                      <div>
                        <button
                          class="MuiButtonBase-root Mui-disabled MuiIconButton-root Mui-disabled MuiIconButton-sizeMedium css-78trlr-MuiButtonBase-root-MuiIconButton-root"
                          data-testid="button-min"
                          disabled=""
                          tabindex="-1"
                          type="button"
                        >
                          <svg
                            aria-hidden="true"
                            class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                            data-testid="RemoveIcon"
                            focusable="false"
                            viewBox="0 0 24 24"
                          >
                            <path
                              d="M19 13H5v-2h14v2z"
                            />
                          </svg>
                        </button>
                        1
                        <button
                          class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium css-78trlr-MuiButtonBase-root-MuiIconButton-root"
                          data-testid="button-plus"
                          tabindex="0"
                          type="button"
                        >
                          <svg
                            aria-hidden="true"
                            class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                            data-testid="AddIcon"
                            focusable="false"
                            viewBox="0 0 24 24"
                          >
                            <path
                              d="M19 13h-6v6h-2v-6H5v-2h6V5h2v6h6v2z"
                            />
                          </svg>
                          <span
                            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                          />
                        </button>
                      </div>
                      <button
                        class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium delete css-78trlr-MuiButtonBase-root-MuiIconButton-root"
                        data-testid="button-delete"
                        tabindex="0"
                        type="button"
                      >
                        <svg
                          aria-hidden="true"
                          class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                          data-testid="DeleteIcon"
                          focusable="false"
                          viewBox="0 0 24 24"
                        >
                          <path
                            d="M6 19c0 1.1.9 2 2 2h8c1.1 0 2-.9 2-2V7H6v12zM19 4h-3.5l-1-1h-5l-1 1H5v2h14V4z"
                          />
                        </svg>
                        <span
                          class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                        />
                      </button>
                    </div>
                  </div>
                </div>
              </div>
              <button
                class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium createButton css-sghohy-MuiButtonBase-root-MuiButton-root"
                data-testid="button-create"
                tabindex="0"
                type="button"
              >
                Buat Pesanan
                <span
                  class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                />
              </button>
            </div>
          </div>
        </div>
      </main>
    </div>
  </body>,
  "container": <div>
    <main
      class="mainWrap"
      data-testid="cart-wrap"
    >
      <div
        class="header"
      >
        <h3>
          Cart
        </h3>
        <button
          class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium backButton css-sghohy-MuiButtonBase-root-MuiButton-root"
          data-testid="button-back"
          tabindex="0"
          type="button"
        >
          <span
            class="MuiButton-startIcon MuiButton-iconSizeMedium css-1d6wzja-MuiButton-startIcon"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
              data-testid="ArrowBackIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M20 11H7.83l5.59-5.59L12 4l-8 8 8 8 1.41-1.41L7.83 13H20v-2z"
              />
            </svg>
          </span>
          Kembali
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </button>
      </div>
      <div
        class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root card css-bhp9pd-MuiPaper-root-MuiCard-root"
      >
        <div
          class="MuiCardContent-root cardContent css-46bh2p-MuiCardContent-root"
        >
          <div>
            <div
              class="merchant"
              data-testid="merchant-info"
            >
              <div
                class="imgWrap"
              >
                <div
                  class="MuiAvatar-root MuiAvatar-square MuiAvatar-colorDefault avatar css-18gxrea-MuiAvatar-root"
                >
                  m
                </div>
              </div>
              <div>
                <h4
                  class="merchantInfoHeader"
                >
                  merchant
                </h4>
                <span
                  class="profileInfo"
                >
                  <p
                    class="profileText"
                  >
                    description
                  </p>
                </span>
              </div>
              <button
                aria-label="Pesan"
                class="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorInherit MuiIconButton-sizeMedium chatButton css-zylse7-MuiButtonBase-root-MuiIconButton-root"
                data-mui-internal-clone-element="true"
                tabindex="0"
                type="button"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                  data-testid="AddCommentOutlinedIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M22 4c0-1.1-.9-2-2-2H4c-1.1 0-2 .9-2 2v12c0 1.1.9 2 2 2h14l4 4V4zm-2 13.17L18.83 16H4V4h16v13.17zM13 5h-2v4H7v2h4v4h2v-4h4V9h-4z"
                  />
                </svg>
                <span
                  class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                />
              </button>
            </div>
            <div>
              <div
                class="service"
                data-testid="service-cart"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                  data-testid="DryCleaningIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M19.56 11.36 13 8.44V7c0-.55-.45-1-1-1s-1-.45-1-1 .45-1 1-1 1 .45 1 1h2c0-1.84-1.66-3.3-3.56-2.95-1.18.22-2.15 1.17-2.38 2.35-.3 1.56.6 2.94 1.94 3.42v.63l-6.56 2.92c-.88.38-1.44 1.25-1.44 2.2v.01C3 14.92 4.08 16 5.42 16H7v6h10v-6h1.58c1.34 0 2.42-1.08 2.42-2.42v-.01c0-.95-.56-1.82-1.44-2.21zM18.58 14H17v-1H7v1H5.42c-.23 0-.42-.19-.42-.43 0-.17.1-.32.25-.38l6.75-3 6.75 3c.15.07.25.22.25.39 0 .23-.19.42-.42.42z"
                  />
                </svg>
                <div
                  class="serviceInfoWrap"
                >
                  <div
                    class="serviceInfo"
                  >
                    <p>
                      service
                    </p>
                    <p>
                      RpÂ 2.000
                    </p>
                  </div>
                  <div
                    class="quantity"
                  >
                    <div>
                      <button
                        class="MuiButtonBase-root Mui-disabled MuiIconButton-root Mui-disabled MuiIconButton-sizeMedium css-78trlr-MuiButtonBase-root-MuiIconButton-root"
                        data-testid="button-min"
                        disabled=""
                        tabindex="-1"
                        type="button"
                      >
                        <svg
                          aria-hidden="true"
                          class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                          data-testid="RemoveIcon"
                          focusable="false"
                          viewBox="0 0 24 24"
                        >
                          <path
                            d="M19 13H5v-2h14v2z"
                          />
                        </svg>
                      </button>
                      1
                      <button
                        class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium css-78trlr-MuiButtonBase-root-MuiIconButton-root"
                        data-testid="button-plus"
                        tabindex="0"
                        type="button"
                      >
                        <svg
                          aria-hidden="true"
                          class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                          data-testid="AddIcon"
                          focusable="false"
                          viewBox="0 0 24 24"
                        >
                          <path
                            d="M19 13h-6v6h-2v-6H5v-2h6V5h2v6h6v2z"
                          />
                        </svg>
                        <span
                          class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                        />
                      </button>
                    </div>
                    <button
                      class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium delete css-78trlr-MuiButtonBase-root-MuiIconButton-root"
                      data-testid="button-delete"
                      tabindex="0"
                      type="button"
                    >
                      <svg
                        aria-hidden="true"
                        class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                        data-testid="DeleteIcon"
                        focusable="false"
                        viewBox="0 0 24 24"
                      >
                        <path
                          d="M6 19c0 1.1.9 2 2 2h8c1.1 0 2-.9 2-2V7H6v12zM19 4h-3.5l-1-1h-5l-1 1H5v2h14V4z"
                        />
                      </svg>
                      <span
                        class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                      />
                    </button>
                  </div>
                </div>
              </div>
            </div>
            <button
              class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium createButton css-sghohy-MuiButtonBase-root-MuiButton-root"
              data-testid="button-create"
              tabindex="0"
              type="button"
            >
              Buat Pesanan
              <span
                class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
              />
            </button>
          </div>
        </div>
      </div>
    </main>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
